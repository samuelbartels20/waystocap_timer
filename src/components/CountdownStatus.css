.countdown-status {
  font-size: 20px;
  font-style: italic;
  line-height: 20px;
}
