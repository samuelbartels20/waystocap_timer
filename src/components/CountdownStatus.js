import React from 'react';
import PropTypes from 'prop-types';

import './CountdownStatus.css';

function CountdownStatus({ secondsRemaining, totalSeconds }) {
  let status = '';
  if (secondsRemaining === 0 && totalSeconds > 0) {
    status = "Time's up!";
  } else if (secondsRemaining / totalSeconds < 0.5) {
    status = 'More than halfway there!';
  }

  return <div className="countdown-status">{status}</div>;
}

CountdownStatus.propTypes = {
  secondsRemaining: PropTypes.number.isRequired,
  totalSeconds: PropTypes.number.isRequired,
};

export default CountdownStatus;
