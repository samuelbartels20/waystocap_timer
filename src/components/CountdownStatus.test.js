import React from 'react';
import CountdownStatus from './CountdownStatus';
import { shallow } from 'enzyme';

it('renders initial state', () => {
  const initialState = shallow(
    <CountdownStatus secondsRemaining={0} totalSeconds={0} />
  );
  expect(initialState).toMatchSnapshot();
});

it('renders times up state', () => {
  const initialState = shallow(
    <CountdownStatus secondsRemaining={0} totalSeconds={10} />
  );
  expect(initialState).toMatchSnapshot();
});

it('renders half way state', () => {
  const initialState = shallow(
    <CountdownStatus secondsRemaining={2} totalSeconds={5} />
  );
  expect(initialState).toMatchSnapshot();
});
