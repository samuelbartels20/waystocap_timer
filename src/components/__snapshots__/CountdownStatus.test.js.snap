// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders half way state 1`] = `ShallowWrapper {}`;

exports[`renders initial state 1`] = `ShallowWrapper {}`;

exports[`renders times up state 1`] = `ShallowWrapper {}`;
